modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -67,12 +81,20 @@ public TransportReindexAction(Settings settings, ThreadPool threadPool, ActionFi
         this.scriptService = scriptService;
         this.autoCreateIndex = autoCreateIndex;
         this.client = client;
+        remoteWhitelist = new HashSet<>(REMOTE_CLUSTER_WHITELIST.get(settings));
+    }
+
+    @Inject(optional=true)
+    public void setHttpServerTransport(HttpServerTransport httpTransport) {
```

Comment:
Can't we just inject the `NodeService` in the constructor and use its `info` method to get the local HTTP adress?